MekHQ/src/mekhq/campaign/universe/PlanetarySystem.java

```diff
@@ -410,13 +410,15 @@ public void setSubtype(double subtype) {
      * 
      */
     public Planet getPrimaryPlanet() {
-    	if(primarySlot<1) {
-    		return planets.get(1);
-    	}
-        return planets.get(primarySlot);
+        return planets.get(getPrimaryPlanetPosition());
     }
     
     public int getPrimaryPlanetPosition() {
+    	if(primarySlot<1) {
+    		//if no primary slot (uninhabited system)
+    		//just return first planet
+    		return(1);
```

Comment:
The parenthesis here are odd. There are also mixed tabs and spaces.

Reply:
lol, thats how you do returns in R which is what I have been coding for the actual system generation. Fixed in 666eaab46.